src/sentry_plugins/trello/plugin.py

```diff
@@ -0,0 +1,278 @@
+from __future__ import absolute_import
+
+import re
+from django.conf.urls import url
+from rest_framework.response import Response
+
+
+from sentry.utils.http import absolute_uri
+from sentry.exceptions import PluginError
+from sentry.plugins.bases.issue2 import IssuePlugin2, IssueGroupActionEndpoint
+from sentry_plugins.base import CorePluginMixin
+from .client import TrelloApiClient
+
+
+SETUP_URL = "https://github.com/getsentry/sentry/blob/master/src/sentry_plugins/trello/Trello_Instructions.md"  # NOQA
+
+
+class TrelloPlugin(CorePluginMixin, IssuePlugin2):
+    description = "Create Trello cards on issues"
+    slug = "trello"
+    title = "Trello"
+    conf_title = title
+    conf_key = "trello"
+    auth_provider = None
+    resource_links = [("Trello Setup Instructions", SETUP_URL)] + CorePluginMixin.resource_links
+
+    def get_config(self, project, **kwargs):
+        """
+        Return the configuration of the plugin.
+        Pull the value out of our the arguments to this function or from the DB
+        """
+
+        def get_value(field):
+            initial_values = kwargs.get("initial") or {}
+            if initial_values.get(field):
+                return initial_values[field]
+            return self.get_option(field, project)
+
+        token_config = {
+            "name": "token",
+            "type": "secret",
+            "label": "Trello API Token",
+            "default": None,
+        }
+
+        token_val = get_value("token")
+        if token_val:
+            token_config["required"] = False
+            token_config["prefix"] = token_val[:5]
+            token_config["has_saved_value"] = True
+        else:
+            token_config["required"] = True
+
+        key_val = get_value("key")
+
+        key_config = {
+            "name": "key",
+            "type": "text",
+            "required": True,
+            "label": "Trello API Key",
+            "default": key_val,
+        }
+
+        config = [key_config, token_config]
+        org_value = get_value("organization")
+        include_org = kwargs.get("add_additial_fields") or org_value
+        if key_val and token_val and include_org:
```

Comment:
Where's the

```py
else:
  raise ValueError("Missing X value")
```

part? Are we swallowing this issue on purpose and signalling the caller that with a `None` result? Seems unpythonic to me.

Reply:
@BYK Not having the key or token is a valid configuration if you haven't set up the plugin yet.